packages/core/src/pages/api/preview.ts

```diff
@@ -32,14 +30,36 @@ const pickParam = (req: NextApiRequest, parameter: string) => {
 // TODO: Improve security by disabling CMS preview in production
 const handler: NextApiHandler = async (req, res) => {
   try {
-    const locator = {
+    const locator: any = {
       contentType: pickParam(req, 'contentType'),
       documentId: pickParam(req, 'documentId'),
       versionId: pickParam(req, 'versionId'),
+      releaseId: pickParam(req, 'releaseId'),
+    }
```

Comment:
suggestion: instead of adding all to the `locator` object and then manipulating it using `delete`, what do you think about adding conditionally? 
```suggestion
    const contentType = pickParam(req, 'contentType')
    const documentId = pickParam(req, 'documentId')
    const versionId = pickParam(req, 'versionId')
    const releaseId = pickParam(req, 'releaseId')

    const locator = {
        ...(contentType !== undefined && { contentType }),
        ...(documentId !== undefined && { documentId }),
        ...(versionId !== undefined && { versionId }),
        ...(releaseId !== undefined && { releaseId })
    }
```

or using the `reduce` function:
```suggestion
    const locator = ['contentType', 'documentId', 'versionId', 'releaseId'].reduce((acc, param) => {
      const value = pickParam(req, param)
      if (value !== undefined) acc[param] = value
      return acc
    }, {} as { [key: string]: string })
```

Reply:
I'd agree with one of @lariciamota 's suggestions for making it more immutable. (avoid using the delete)